server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -903,6 +906,53 @@ public void testSimpleOperations() throws Exception {
         searchResult.close();
     }
 
+    public void testGetWithSearcherWrapper() throws Exception {
+        engine.refresh("warm_up");
+        engine.index(indexForDoc(createParsedDoc("1", null)));
+        assertThat(engine.lastRefreshedCheckpoint(), equalTo(NO_OPS_PERFORMED));
+        try (Engine.GetResult get = engine.get(new Engine.Get(true, true, "1"), docMapper(), randomSearcherWrapper())) {
+            // we do not track the translog location yet
+            assertTrue(get.exists());
+            assertFalse(get.isFromTranslog());
+        }
+        // refresh triggered, as we did not track translog location until the first realtime get.
+        assertThat(engine.lastRefreshedCheckpoint(), equalTo(0L));
+
+        engine.index(indexForDoc(createParsedDoc("1", null)));
+        try (Engine.GetResult get = engine.get(new Engine.Get(true, true, "1"), docMapper(), searcher -> searcher)) {
+            assertTrue(get.exists());
+            assertTrue(get.isFromTranslog());
+        }
+        assertThat(engine.lastRefreshedCheckpoint(), equalTo(0L)); // no refresh; just read from translog
+        if (randomBoolean()) {
+            engine.index(indexForDoc(createParsedDoc("1", null)));
+        }
+        try (Engine.GetResult get = engine.get(new Engine.Get(true, true, "1"), docMapper(),
+            searcher -> SearcherHelper.wrapSearcher(searcher, reader -> new MatchingDirectoryReader(reader, new MatchAllDocsQuery())))) {
+            assertTrue(get.exists());
+            assertFalse(get.isFromTranslog());
```

Comment:
`GetResult.isFromTranslog` is slightly confusing now, in that in this case, the value is really taken from translog, but still we return false here. I think we should add a comment to `isFromTranslog` to clarify this in detail to avoid someone relying on the method, expecting a different meaning.

Reply:
++, added in https://github.com/elastic/elasticsearch/pull/64504/commits/792ab09a8cc8692898d334a2a317b8a33bd39e3a